apps/emqx_resource/src/emqx_resource_manager.erl

```diff
@@ -605,21 +608,27 @@ update_state(Data, _DataWas) ->
 health_check_interval(Opts) ->
     maps:get(health_check_interval, Opts, ?HEALTHCHECK_INTERVAL).
 
-maybe_alarm(connected, _ResId, _Error) ->
+maybe_alarm(connected, _ResId, _Error, _PrevError) ->
     ok;
-maybe_alarm(_Status, <<?TEST_ID_PREFIX, _/binary>>, _Error) ->
+maybe_alarm(_Status, <<?TEST_ID_PREFIX, _/binary>>, _Error, _PrevError) ->
     ok;
-maybe_alarm(_Status, ResId, Error) ->
+maybe_alarm(_Status, ResId, Error, none = _PrevError) ->
     HrError =
         case Error of
             undefined -> <<"Unknown reason">>;
             _Else -> emqx_utils:readable_error_msg(Error)
         end,
-    emqx_alarm:activate(
-        ResId,
-        #{resource_id => ResId, reason => resource_down},
-        <<"resource down: ", HrError/binary>>
-    ).
+    ?SAFE_CALL_WITH_LOG(
```

Comment:
move the safety code to the callee module

Reply:
Done.